packages/application-shell/src/components/application-shell/helpers.ts

```diff
@@ -34,3 +36,52 @@ export function getMcOrigin(mcApiUrl: string, actualWindow = window) {
 
   return mcApiUrlObject.origin.replace('mc-api', 'mc');
 }
+
+const defaultAsyncMessagesLoader =
+  (availableLanguages: string[]) => (lang: string) => {
+    const isAvailableLanguage = availableLanguages?.includes(lang);
+
+    return import(
+      `../../i18n/data/${
```

Comment:
I feel like this is the main challenge to figure out. Maybe we can do something like this, using the `webpack.DefinePlugin`

<img width="1117" alt="Screen Shot 2022-01-24 at 09 37 44" src="https://user-images.githubusercontent.com/1110551/150748967-6eb72b7e-0b32-4429-88e7-efe2f13975db.png">

Reply:
Well, I think we have a problem here with my initial expectations because of the way [dynamic expressions](https://webpack.js.org/api/module-methods/#dynamic-expressions-in-import) work in Webpack dynamic imports.

It is not possible to use a complete dynamic path because webpack needs to know about the possible files it would need to lazy load. So you give it a relative path and it creates a bundle for every possible combinations.

As the `ApplicationShell` component is imported from the consumer application, I don't seem to find the right path to use in the different situations (locally development in the app-kit, locally development from the mc-frontent and production both default and custom apps).

The only idea I came up with is to delegate to the consumer app the call to the dynamic `import` so it can correctly set the translations folder and leave it to the `ApplicationShell` to figure out which file should be loaded.

From the `ApplicationShell` perspective, it would look like this:
```
const EntryPoint = () => (
  <ApplicationShell
    environment={window.app}
    localeLoader={(localeFileName) =>
      import(
        `../../i18n/data/${localeFileName}` /* webpackChunkName: "app-i18n-[request]" */
      )
    }
    onRegisterErrorListeners={({ dispatch }) => {
      Sdk.Get.errorHandler = (error) =>
        globalActions.handleActionError(error, 'sdk')(dispatch);
    }}
    render={() => <ApplicationStateMachines />}
  />
);
```

I'm not very proud of this solution, but it's the only I currently have.

Regarding the proposed ```availableLocales``` prop, after thinking about it twice, it doesn't seem too valuable to me.
It is only used to validate the language requested is configured by the developers of Apps, but the component would anyway return the default translations (english) so I don't see the point of having it actually.

I pushed a [commit}(https://github.com/commercetools/merchant-center-application-kit/pull/2452/commits/03f75350531e8c8abf983a5dd85bb5daf5d3e90e) with the implementation for this idea to better explaining but I wouldn't like to go further before getting some feedback about it.

Reply:
So it didn't work when using `webpack.DefinePlugin`?

Reply:
Ah but I guess the problem is also when we bundle the packages. Rollup will find dynamic imports for code splitting but we actually don't want to execute it just yet.

So yeah, I guess using the `import()` syntax can only be done in the application code itself.

There was actually another small issue that I forgot to mention. The translation files are usually located in `i18n/data`, however we also use the option to get the "compiled" version of the translation object which are located in `i18n/compiled-data`.

At this point I feel like we're going down a bit the rabbit hole, as there are no more clear benefits with a new approach.

Let's have a chat about this but I suppose we should put on hold this for now.